spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -168,19 +172,105 @@ public void setExtendedBindingProperties(RabbitExtendedBindingProperties extende
 	}
 
 	@Override
-	public void onInit() {
+	public void onInit() throws Exception {
+		super.onInit();
+
+		CachingConnectionFactory producerConnectionFactory = createProducerConnectionFactory(this.rabbitProperties);
+		producerConnectionFactory.setApplicationContext(getApplicationContext());
+		getApplicationContext().addApplicationListener(producerConnectionFactory);
+		producerConnectionFactory.afterPropertiesSet();
+
```

Comment:
Hmmm - how about 

    if (this.connectionFactory instanceof CachingConnectionFactory) {
        this.producerConnectionFactory = new CCF(((CCF) this.connectionFactory).getRabbitConnectionFactory()).
    }

We should probably add that getter to the interface, perhaps in S-A 2.0.